docs/external-references.md

```diff
@@ -18,27 +18,78 @@ MyST supports referencing rich content in a growing number of formats, including
+
+(myst-xref-config)=
+
+```yaml
+references:
+  myst-guide: https://mystmd.org/guide
+  jtex: https://mystmd.org/jtex
+```
+
+When you specify these in your project configuration, MyST will load the remote `myst.xref.json` file,
+and provide access to all of the pages and available reference targets in that project. This `json` file will be cached to disk locally in the `_build` folder, eliminating duplicate web requests on subsequent builds. If the target documentation updates and you must reload the remote references, you may delete the cache with `myst clean --cache`.
+
+````{important}
+# MyST Cross Reference Examples
+
+```{list-table}
+:header-rows: 1
+* - MyST Syntax
+  - Rendered
+* - `[](xref:myst-guide)`
+    : A reference to the first page of the MyST Guide.
+  - [](xref:myst-guide)
+* - `[](xref:myst-guide#frontmatter-before)`
+    : A reference to a labeled figure in the MyST Guide.
+  - [](xref:myst-guide#frontmatter-before)
+* - `<xref:jtex/template-rules#syntax>`
+    : Alternate link syntax that references a heading on a specific page of the jtex documentation.
+  - <xref:jtex/template-rules#syntax>
 ```
+````
+
+To reference content in the linked MyST project, use the `xref:` protocol in a link followed by the `project` key (from `references`), the url path and the target.
+For example, `<xref:jtex/template-rules#syntax>` renders to:\
+"<xref:jtex/template-rules#syntax>"\
+and is made of a `protocol`, `project`, `path` and `target`.
+
+protocol
+: The protocol for this type of link is `xref:`, and is what selects for cross-project referencing.
+
+project
+: the `project` key above is "jtex" which is defined in your local [project configuration](#myst-xref-config) above.
+
+path
+: the `path` is everything that follows the project before the `#`. It corresponds directly to the URL path in the MyST site. In the example above, path is "/template-rules"
+: Path is optional in most cases; the `target` can be resolved against the entire project without the path. An exception to this is headings without an explicit label in the source markdown - these will require path.
+
+target
+: The target is everything that follows the `#` and is a named reference in the project. In the example above it is "syntax".
+: Target is also optional - if not provided, the cross reference will simply link to a page in the external MyST project. However, without the target, there will not be a rendered tooltip.
+: When creating a cross reference, you can determine `path` and `target` by simply navigating to your target content in the external MyST project and copying the path and fragment from the URL.
+
+If no link text is provided, e.g. `[](xref:...)`, text will be generated from the external project at build-time. You may of course override this behavior by providing you own text, `[text](xref:...)`.
```

Comment:
````suggestion
If no link text is provided, e.g. `[](xref:...)`, text will be generated from the external project at build-time. You may of course override this behavior by providing you own text, `[text](xref:...)`.

```{tip} References are stored in a `myst.xref.json` file

All MyST sites published to the web expose a `myst.xref.json` file that contains all of the cross-references information about a MyST site.
This file is what MyST pulls when you point to an external MyST site for cross-referencing.
It is also a machine-readable record that can be used for analyzing the cross-referencing behavior of MyST projects.
```
````

Reply:
I tried to move conversation about the `.json` blob to the end of this section, since it feels more like an "expert feature" to me that isn't immediately relevant for understanding how to reference. I figure putting it at the end helps separate out the topics a bit more.

Reply:
Nice, moved and merged.